src/fs/fuse.rs

```diff
@@ -38,1190 +38,557 @@ const U64_SIZE: usize = ::core::mem::size_of::<u64>();
 const S_IFLNK: u32 = 40960;
 const S_IFMT: u32 = 61440;
 
-#[allow(dead_code)]
-const FUSE_GETATTR_FH: u32 = 1 << 0;
-
-#[repr(C)]
-#[derive(Debug)]
-struct fuse_dirent {
-	pub d_ino: u64,
-	pub d_off: u64,
-	pub d_namelen: u32,
-	pub d_type: u32,
-	pub d_name: [u8; 0],
-}
-
 pub(crate) trait FuseInterface {
-	fn send_command<S, T>(&mut self, cmd: &Cmd<S>, rsp: &mut Rsp<T>)
-	where
-		S: FuseIn + core::fmt::Debug,
-		T: FuseOut + core::fmt::Debug;
+	fn send_command<const CODE: u32>(
+		&mut self,
+		cmd: &<Op<CODE> as OpTrait>::Cmd,
+		rsp: &mut <Op<CODE> as OpTrait>::Rsp,
+	) where
+		Op<CODE>: OpTrait;
 
 	fn get_mount_point(&self) -> String;
 }
 
-#[repr(C)]
-#[derive(Debug, Default)]
-struct fuse_in_header {
-	pub len: u32,
-	pub opcode: u32,
-	pub unique: u64,
-	pub nodeid: u64,
-	pub uid: u32,
-	pub gid: u32,
-	pub pid: u32,
-	pub padding: u32,
-}
-
-#[repr(C)]
-#[derive(Default, Debug)]
-struct fuse_out_header {
-	pub len: u32,
-	pub error: i32,
-	pub unique: u64,
-}
+/// Marker trait, which signals that a struct is a valid Fuse command.
+/// Struct has to be repr(C)!
+pub(crate) unsafe trait FuseIn {}
+/// Marker trait, which signals that a struct is a valid Fuse response.
+/// Struct has to be repr(C)!
+pub(crate) unsafe trait FuseOut {}
```

Comment:
We'd need a safety comment on these, explaining exactly what is structurally depended upon. Is there any overlap with [`zerocopy::FromBytes`](https://docs.rs/zerocopy/latest/zerocopy/trait.FromBytes.html)?

Reply:
I think we can also simply remove them. They were there in the previous version of the code and I kept them since I thought they did not hurt but I don't think they are all that useful anymore since we clarify the input/output relationship with the associated types.

Reply:
By the way, I removed the traits in 62f08f61548a6aa7216324b871f1ee67ffbfd1e1.